### ISSUE TITLE
Carrier configuration for "showmethod" does not appear in carrier configuration

### ISSUE THREAD

**Issue by matthew-muscat** [magento/magento2#5693]
The carrier settings for the "showmethod" configuration field does not appear for both the standard Magento Shipping Methods (ie: free shipping / flat rate) as well as custom shipping methods.
## Steps to reproduce
1. Install Magento CE 2.1.0.
2. Review the "Free Shipping" / "Table Rate" and/or "Flat Rate" method settings in "Stores / System / Configuration / Shipping Methods".
3. The setting "showmethod" (with the label "Show Method if Not Applicable") does not display, despite being specified in the "Free Shipping" method's adminhtml/system.xml file. @see https://github.com/magento/magento2/blob/f10c7c83a93628122a929949b97e08faa13ce652/app/code/Magento/OfflineShipping/etc/adminhtml/system.xml#L147
## Expected result

The "Show Method if Not Applicable" setting should appear for the above carriers, as well as custom carriers.
## Actual result

The "Show Method if Not Applicable" setting does not appear for the above carriers, as well as custom carriers.
- Changing the value of the "showmethod" field to "showmethod_test" allows the field to display - Is there some other logic tied to the display of this field in the admin area?
## Sample code from custom carrier system.xml

```
<field id="showmethod" translate="label" type="select" sortOrder="170" showInDefault="1" showInWebsite="1" showInStore="0">
    <label>Show Method if Not Applicable</label>
    <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
</field>

<field id="specificerrmsg" translate="label" type="textarea" sortOrder="180" showInDefault="1" showInWebsite="1" showInStore="1">
    <label>Displayed Error Message</label>
    <depends>
        <field id="showmethod">1</field>
    </depends>
</field>
```

**Comment by viktym** [comment]
Thanks for reporting this issue @matthew-muscat 
We've created internal ticket MAGETWO-59181 regarding this

**Comment by SerhiyShkolyarenko** [comment]
The fix is delivered to develop branch. Closing the issue.

**Comment by Ctucker9233** [comment]
@SerhiyShkolyarenko Will MAGETWO-59185 be included in 2.1.3?

**Comment by dcabrejas** [comment]
@SerhiyShkolyarenko can you confirm what version of Magento has this fix? I am having the same problem on 2.1.7.
